packages/financial-templates-lib/src/price-feed/QuandlPriceFeed.js

```diff
@@ -0,0 +1,248 @@
+      this.logger.debug({
+        at: "QuandlPriceFeed",
+        message: "Update skipped because the last one was too recent",
+        currentTime: currentTime,
+        lastUpdateTimestamp: this.lastUpdateTime,
+        timeRemainingUntilUpdate: this.lastUpdateTimes + this.minTimeBetweenUpdates - currentTime
+      });
+      return;
+    }
+
+    this.logger.debug({
+      at: "QuandlPriceFeed",
+      message: "Updating QuandlPriceFeed",
+      currentTime: currentTime,
+      lastUpdateTimestamp: this.lastUpdateTime
+    });
+
+    // Find the closest day that completed before the beginning of the lookback window, and use
+    // it as the start date.
+    const startLookbackWindow = currentTime - this.lookback;
+    const startDateString = this._secondToDateTime(startLookbackWindow);
+    const endDateString = this._secondToDateTime(currentTime);
+
+    // 1. Construct URL.
+    // See https://docs.quandl.com/docs/parameters-2 for how this url is constructed.
+    const url = [
+      `https://www.quandl.com/api/v3/datasets/${this.datasetCode}/${this.databaseCode}/data.json?`,
+      `start_date=${startDateString}&end_date=${endDateString}`,
+      `&collapse=daily&api_key=${this.apiKey}`
+      // Theoretically you could change granularity to be greater than daily but this doesn't seem
+      // useful to implement flexibility for right now.
+    ].join("");
+
+    // 2. Send request.
+    const historyResponse = await this.networker.getJson(url);
+
+    // 3. Check responses.
+    if (
+      !historyResponse?.dataset_data?.data ||
+      historyResponse.dataset_data.data.length === 0 ||
+      historyResponse.dataset_data.data.some(dailyData => dailyData.length === 0)
+    ) {
+      throw new Error(`🚨Could not parse price result from url ${url}: ${JSON.stringify(historyResponse)}`);
+    }
+
+    // 4. Parse results.
+    // Return data structure:
+    // {
+    //   "dataset_data": {
+    //     "data": [
+    //       [
+    //         "2021-03-16",
+    //          1730.2,
+    //          1740.5,
+    //          1724.4,
+    //          1730.1,
+    //          1.7,
+    //          1730.9,
+    //          41843.0,
+    //          21285.0
+    //       ]
+    //      ...more data for different days
+    //     ],
+    //     ...other data we don't care about
+    //   }
+    // }
+    const newHistoricalPricePeriods = historyResponse.dataset_data.data
+      .map(dailyData => ({
+        // Output data should be a list of objects with only the open and close times and prices.
+        // Note: Data is formatted as [Date, Open, High, Low, Last, Change, Settle, Volume, Previous Day Open Interest]
+        openTime: this._dateTimeToSecond(dailyData[0]) - 24 * 3600,
+        closeTime: this._dateTimeToSecond(dailyData[0]),
+        openPrice: this.convertPriceFeedDecimals(dailyData[1]),
+        closePrice: this.convertPriceFeedDecimals(dailyData[4])
```

Comment:
probably not a huge deal, but it might make it easier (maybe) to use this data if its in the form 
```
[
  [time,price]
]
```
Its slightly more portable (easier to implement getHistoricalPricePeriods) and I dont think theres anywhere currently that requires this olhc shape.

You could do something like this to capture both open and close times, though your lookup function would need to change:

```js
.map(({openTime,openPrice,...etc})=>{
  return [
    [openTime,openPrice],
    [closeTime,closePrice],
  ]
}).flat() // should flatten this structure out to an array of [time,prices]
```

wdyt?

Reply:
So I do think we should standardize how price periods are stored in all price feeds to `[time,price]`. For now I'm going to hold off since `getHistoricalPricePeriods = () => this.historicalPricePeriods.map(ohlc => [ohlc.closeTime, ohlc.closePrice])` is easy enough to implement right now. But once we re-evaluate the `update()` and `getPrice`() architecture, we're going to change a lot of the internal data structures and move towards a standard interface. Good reminder